src/ann_solo/reader.py

```diff
@@ -231,6 +295,233 @@ def get_version(self) -> str:
         """
         return 'null'
 
+    def _parse_sptxt_spectrum(self, raw_spectrum: str, identifier: int = None)
+        -> MsmsSpectrum:
+        """
+        Takes a raw spectrum data retrieved from an sptxt file and
+        parses it to a structured object of type MsmsSpectrum.
+
+        Parameters
+        ----------
+        raw_spectrum : string
+            The spectrum in a raw format.
+        identifier : int
+            Incremented identifier of the spectrum in the library.
+
+        Returns
+        -------
+        MsmsSpectrum
+            An MsmsSpectrum object.
+        """
+        # Create a reference variable for Class RegexpTokenizer
+        # that split raw spectrum in two chunks: metadata & spectrum
+        spectrum_tk = RegexpTokenizer('Num\s?Peaks:\s?[0-9]+\n', gaps=True,
+                                      flags=re.IGNORECASE)
+        spectrum_tokens = spectrum_tk.tokenize(raw_spectrum)
+        spectrum_metadata = spectrum_tokens[0]
+        spectrum = spectrum_tokens[1]
+        ##Retrieve identifier if there is one
+        LibID = re.search('LibID:\s?[0-9]+', spectrum_metadata, re.IGNORECASE)
+        if LibID:
+            identifier = re.search('[0-9]+.', LibID.group(0)).group(0)
+        # Check if decoy
+        decoy = True if re.search('decoy', spectrum_metadata,
+                                  re.IGNORECASE) else False
+        ##Retrieve peptide & charge
+        peptide_Charge = spectrum_metadata.split('\n', 1)[0].split('/')
+        peptide = peptide_Charge[0].strip()
+        charge = int(peptide_Charge[1].strip())
+        ##Retrieve precurssor mass
+        perMZ = re.search('PrecursorMZ:\s?[0-9]+.[0-9]+', spectrum_metadata,
+                          re.IGNORECASE)
+        if perMZ:
+            perMZ = re.search('[0-9]+.[0-9]+', perMZ.group(0))
+        else:
+            perMZ = re.search('Parent=\s?[0-9]+.[0-9]+', spectrum_metadata,
+                              re.IGNORECASE)
+            perMZ = re.search('[0-9]+.[0-9]+', perMZ.group(0))
+        ##Retrieve MZ & Intensities
+        file = io.StringIO(spectrum)
+        MZ_Intensity = pd.read_csv(file, sep="\t", header=None)
```

Comment:
**suggestion:** Can we parse the peaks using NumPy instead of Pandas for increased efficiency?